File: leetcode-ts/src/no0001.test.ts
import {expect} from 'chai';

import {No0001} from './no0001';

describe('no0001 test', function () {
    it('(example01) should return two number for the target', () => {
        const no0001 = new No0001();
        expect(no0001.twoSum([2, 7, 11, 15], 9)).deep.equal([1, 0])
    });

    it('(example02) should return two number for the target', () => {
        const no0001 = new No0001();
        expect(no0001.twoSum([3, 2, 4], 6)).deep.equal([2, 1])
    });

    it('(example03) should return two number for the target', () => {
        const no0001 = new No0001();
        expect(no0001.twoSum([3, 3], 6)).deep.equal([1, 0])
    });
});

File: leetcode/src/test/java/com/github/lucaseo90/ds/day01/array/No0001Test.java
package com.github.lucaseo90.ds.day01.array;

import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;

public class No0001Test {

    private No0001 no0001 = new No0001();

    @Test
    public void example01() {
        Assert.assertEquals("[0, 1]", Arrays.toString(no0001.twoSum(new int[]{2, 7, 11, 15}, 9)));
    }

    @Test
    public void example02() {
        Assert.assertEquals("[1, 2]", Arrays.toString(no0001.twoSum(new int[]{3, 2, 4}, 6)));
    }

    @Test
    public void example03() {
        Assert.assertEquals("[0, 1]", Arrays.toString(no0001.twoSum(new int[]{3, 3}, 6)));
    }
}
